### ISSUE TITLE
About dilated convolution

### ISSUE THREAD

**Issue by Hu01001** [saeed-anwar/DRLN#3]
For feature map with 1x1xc after AdaptiveAvgPool2d,why can dilated convolution with different dilation rate make different results? I have a question about this.

**Comment by f4hxm** [comment]
I have the same question, because there is a sentence "zeros are padded accordingly" in the paper , which means  dilated convolutions with different dilation rates make the same results.

**Comment by rajesh1226** [comment]
"zeros are padded " accordingly is implemnted in the following lines in CALayer. Hope this helps.

    self.c1 = ops.BasicBlock(channel , channel // reduction, 3, 1, 3, 3)
     self.c2 = ops.BasicBlock(channel , channel // reduction, 3, 1, 5, 5)
     self.c3 = ops.BasicBlock(channel , channel // reduction, 3, 1, 7, 7)

**Comment by GuideWsp** [comment]
the so-called "laplacian attention" is just "SE" attention.

**Comment by miracleyoo** [comment]
Yes totally agree. It confused me a lot about the so-called "Laplacian attention" when I looked into the code, all three dilated Conv is actually doing nothing different at all since, except the central point, all of the other points are all zero...... I doubt this paper's sincerity and honor.

**Comment by saeed-anwar** [comment]
The padding is done to the patch, not to the kernels. Dilated Convs are working on the same patch at the same locations but with different dilations, hence each one using different features of the patch. Hope this helps.

**Comment by kelvinkoh0308** [comment]
@saeed-anwar This still does not make sense because only the center location of the weight in the convolution kernel will be involved in the output because all others weight will be multiplied with the zero value of the padding. This means that we initialize extra weight in the convolution kernel but only using one weight in the kernel and the dilations also do not change the fact that only one weight will be used.

![image](https://user-images.githubusercontent.com/50705244/94342125-6c03e180-0041-11eb-8183-9449a16375f9.png)

As you can see from the example above, the outputs only involve the weight in the center location of the kernel. If this is the case, why not we just use 1x1 convolution to replace the dilated convolution which is more efficient?

**Comment by JingshuaiLiu** [comment]
@miracleyoo, @kelvinkoh0308, totally agree. 

I suppose that the features g_d, which have been reduced to 1✕1, will be padded to be of the same size of the dilated kernel, when the configuration is such that kernel_size=3 and padding=dilation. Hence, the parameters only in the centre will take potentially non-zero values, i.e. the pooled 1✕1 information. Therefore, can we say that the Laplacian pyramid attention module takes two additional parallel MLP paths to determine the adaptive weights of the initial feature maps?
